### PR TITLE
Add floating point division in PageStatistics

### LINKED ISSUE #399: Convert integer division to floating point division (PageStatistics)
**Context**

In `calculateAveragePageLength()` in `PageStatistics.java`, we have the following:

```java
    public Integer calculateAveragePageLength() {
        // ...
        return (booksWithPagesSpecified == 0) ? 0 :
                (int) Math.ceil(totalNumberOfPages / booksWithPagesSpecified);
    }
```

Both `totalNumberOfPages` and `booksWithPagesSpecified` are ints. Hence, the cast to an `int` is redundant. 

**What we want**

- Cast one of the operands to a float so that we can first get the result, round up and then truncate.
- Add an accompanying test to ensure this works as intended

### PR DESCRIPTION
**Pull request by ShreyaPrabhu**
## Summary of change
Add float point division in Page Statistics file to the average page length calculation function.

## Additional context
Added a test case as well to verify above mentioned change. 
Test case description - 
Initial setup -
Total Number of pages = 1750 
Total Number of books = 7
I deleted one book with number of pages 150 to get following new values - 
Total Number of pages =1600 
Total Number of books = 6
Division result = 266.66 
After Math.ceil(266.66) = 267

I have made changes to clear savedBooks list and reset totalRating to 0.0 in populateReadBooks function under StatisticTestUtils.java file. I noticed that every time bookService.deleteAll() was called, though it cleared the repository, it did not clear the list or reset total rating which is expected behavior. 

Ran all test cases to make sure no other test cases failed due to the changes made in this PR.
Ran checkstyle validation after code changes.

[If a screenshot would be useful (e.g. for UI changes), please provide one.]

## Related issue

Closes #399 

## Pull request checklist

Please keep this checklist in & ensure you have done the following:

- [x] Read, understood and adhered to our [contributing document](https://github.com/knjk04/book-project/blob/master/CONTRIBUTING.md).
  - [x] Ensure that you were first assigned to a relevant issue before creating this pull request
  - [x] Ensure code changes pass all tests

- [x] Read, understood and adhered to our [style guide](https://github.com/knjk04/book-project/blob/master/STYLEGUIDE.md). A lot of our code reviews are spent on ensuring compliance with our style guide, so it would save a lot of time if this was adhered to from the outset. 

- [x] Filled in the summary, context (if applicable) and related issue section. Replace the square brackets and its placeholder content with your contents. For an example, see any merged in pull request
  - [x] Included a screenshot(s) if a UI change was involved (it may look different on a reviewer's device)

- [x] Created a branch that has a descriptive name (what your branch is for in a few words and includes the issue number at the end, e.g. `test-reading-goal-123`

- [x] Set this pull request to 'draft' if you are still working on it

- [x] Resolved any merge conflicts

If in doubt, get in touch with us via our [Slack workspace](https://join.slack.com/t/teambookproject/shared_invite/zt-h6vx3n8l-KN8QnO50r7QWZHgHFnFdPw)

### CODE REVIEW

**Comment by ShreyaPrabhu** [comment]
@knjk04 This PR is ready for review. I was not able to add reviewers or mark the PR as ready to review. Hence commenting here.

**Comment by knjk04** [comment]
@ShreyaPrabhu Thanks, we'll review it shortly

File: src/main/java/com/karankumar/bookproject/backend/statistics/PageStatistics.java [hunk 1]
```diff
@@ -67,6 +67,6 @@ public Integer calculateAveragePageLength() {
             return null;
         }
         return (booksWithPagesSpecified == 0) ? 0 :
-                (int) Math.ceil(totalNumberOfPages / booksWithPagesSpecified);
+                (int) Math.ceil(totalNumberOfPages / (float) booksWithPagesSpecified);
```
Comment (knjk04) [inline src/main/java/com/karankumar/bookproject/backend/statistics/PageStatistics.java:70]: Could you please remove the `int` cast? I think it's better to return float here (sorry, I should have mentioned this in the issue)

File: src/test/java/com/karankumar/bookproject/backend/statistics/PageStatisticsTest.java [hunk 2]
```diff
@@ -89,6 +89,14 @@ void averagePageLengthExistsAndIsCorrect() {
         assertEquals(averagePageLength, pageStatistics.calculateAveragePageLength());
     }
 
+    @Test
+    void averagePageLengthWhenRoundedToHigherNumber() {
+        StatisticTestUtils.deleteBook(StatisticTestUtils.getBookWithHighestRating());
+        pageStatistics = new PageStatistics(predefinedShelfService);
+        int averagePageLength = (int) Math.ceil(StatisticTestUtils.getTotalNumberOfPages() / (float) StatisticTestUtils.getNumberOfBooks());
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/backend/statistics/PageStatisticsTest.java:96]: Could you please ensure this line adheres to our 100 character column limit?

File: src/test/java/com/karankumar/bookproject/backend/statistics/PageStatisticsTest.java [hunk 3]
```diff
@@ -89,6 +89,14 @@ void averagePageLengthExistsAndIsCorrect() {
         assertEquals(averagePageLength, pageStatistics.calculateAveragePageLength());
     }
 
+    @Test
+    void averagePageLengthWhenRoundedToHigherNumber() {
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/backend/statistics/PageStatisticsTest.java:93]: Could you please rename this method to better reflect what is being tested?

File: src/test/java/com/karankumar/bookproject/backend/statistics/PageStatisticsTest.java (hunk 2 above)
Comment (knjk04) [reply src/test/java/com/karankumar/bookproject/backend/statistics/PageStatisticsTest.java:96]: Also, could you please replace this line with a hardcoded value? I think the problem with the current approach is that we're redefining the code under test, so this should always pass.

File: src/test/java/com/karankumar/bookproject/backend/statistics/utils/StatisticTestUtils.java [hunk 4]
```diff
@@ -117,4 +126,12 @@ public static int getTotalNumberOfPages() {
         }
         return pages;
     }
+
+    public static void deleteBook(Book bookToDelete) {
+        PredefinedShelf readShelf = predefinedShelfUtils.findReadShelf();
+        bookToDelete.setPredefinedShelf(readShelf);
```
Comment (knjk04) [inline src/test/java/com/karankumar/bookproject/backend/statistics/utils/StatisticTestUtils.java:132]: I don't think these first two lines are needed

**Review by knjk04 — CHANGES_REQUESTED** [review]
Thanks for working on this! I've left some feedback for improvements

**Comment by knjk04** [comment]
Let us know when you're ready for another review

**Comment by ShreyaPrabhu** [comment]
@knjk04 This is ready for review. Have made the changes requested.

**Comment by knjk04** [comment]
@ShreyaPrabhu Could you please resolve the merge conflict?

**Comment by ShreyaPrabhu** [comment]
@knjk04 Merge conflict was present in one file. It has been resolved.

**Review by knjk04 — APPROVED** [review]

**Comment by knjk04** [comment]
@ShreyaPrabhu Thank you for working on this!

**Pull request merged**